src/pip/_internal/metadata/importlib.py

```diff
@@ -139,25 +162,11 @@ def version(self) -> DistributionVersion:
     def is_file(self, path: InfoPath) -> bool:
         return self._dist.read_text(str(path)) is not None
 
-    def iterdir(self, path: InfoPath) -> Iterator[pathlib.PurePosixPath]:
-        # We have an optimized implementation for wheels.
-        with contextlib.suppress(AttributeError):
-            return self._dist.iterdir(path)  # type: ignore [attr-defined]
-
-        # This is not actually based on anything concrete to iterate on.
-        if self._info_location is None:
-            raise FileNotFoundError(path)
-
-        # Just use pathlib if this is actually on the filesystem.
-        info_root = self._info_location
-        full_path = info_root.joinpath(path)
-        if isinstance(full_path, pathlib.Path):
-            return (
-                pathlib.PurePosixPath(p.relative_to(info_root).as_posix())
-                for p in full_path.iterdir()
-            )
-
-        raise FileNotFoundError(path)  # Nothing else is implemetned for now.
+    def iter_distutils_script_names(self) -> Iterator[str]:
+        if not isinstance(self._info_location, pathlib.Path):
+            return
+        for child in self._info_location.joinpath("scripts").iterdir():
```

Comment:
A comment on why `self._info_location / "scripts"` is a right directory would be useful.

Reply:
This is already explained in `BaseDistribution`.